validator/src/main.rs

```diff
@@ -838,6 +848,16 @@ pub fn main() {
         exit(1);
     });
 
+    let mut snapshot_compression = CompressionType::Bzip2;
+    if let Ok(compression_str) = value_t!(matches, "snapshot_compression", String) {
+        match compression_str.as_str() {
+            "bz2" => snapshot_compression = CompressionType::Bzip2,
+            "gzip" => snapshot_compression = CompressionType::Gzip,
+            "zstd" => snapshot_compression = CompressionType::Zstd,
+            "none" => snapshot_compression = CompressionType::NoCompression,
+            _ => panic!("Compression type not recognized: {}", compression_str),
+        }
```

Comment:
Use https://docs.rs/clap/2.33.0/clap/struct.Arg.html#method.possible_values to remove the panic! here.  User input should never cause a panic

Reply:
I added the possible_values so this should never be hit.